scripts/python/src/fodt/helpers.py

```diff
@@ -30,6 +30,28 @@ def create_backup_document(filename) -> None:
         backup_file = backupdir / filename.name
         return backup_file
 
+    @staticmethod
+    def derive_maindir_from_filename(filename: str) -> Path:
+        """filename is a assumed to be an aboslute path to file inside maindir or
+        subdirectories of maindir."""
+        filename = Path(filename)
+        assert filename.is_absolute()
+        # Search parent directories for main.fodt in a directory called "parts"
+        while True:
+            # Check if we have reached the root directory
+            #  filename.parent == filename is True if filename is the root directory
+            if filename.parent == filename:
```

Comment:
Would it be possible to change this and stop in the directory where .git is or the root director<?